### PR TITLE
chore(bench): add benchmarks for integer oprf

### PR DESCRIPTION
**Pull request by soonum**
closes: zama-ai/tfhe-rs-internal#564

### CODE REVIEW

**Review by mayeul-zama — APPROVED** [review]
Thanks!

**Comment by IceTDrinker** [comment]
cargo build is red

File: .github/workflows/integer_full_benchmark.yml [hunk 1]
```diff
@@ -106,6 +106,7 @@ jobs:
       - name: Run benchmarks with AVX512
         run: |
           make BENCH_OP_FLAVOR=${{ matrix.op_flavor }} bench_${{ matrix.command }}
+          make bench_integer_oprf
```
Comment (IceTDrinker) [inline .github/workflows/integer_full_benchmark.yml:109]: ça risque de lancer le bench oprf beaucoup de fois non ?

File: Makefile [hunk 2]
```diff
@@ -832,23 +832,26 @@ bench_integer_zk: install_rs_check_toolchain
...
 .PHONY: bench_shortint # Run benchmarks for shortint
 bench_shortint: install_rs_check_toolchain
 	RUSTFLAGS="$(RUSTFLAGS)" __TFHE_RS_BENCH_OP_FLAVOR=$(BENCH_OP_FLAVOR) \
 	cargo $(CARGO_RS_CHECK_TOOLCHAIN) bench \
 	--bench shortint-bench \
 	--features=$(TARGET_ARCH_FEATURE),shortint,internal-keycache,nightly-avx512 -p $(TFHE_SPEC)
 
-.PHONY: bench_oprf # Run benchmarks for shortint
-bench_oprf: install_rs_check_toolchain
+.PHONY: bench_shorint_oprf # Run benchmarks for shortint
```
Comment (IceTDrinker) [inline Makefile:849]: typo shorint -> shortint

File: .github/workflows/integer_full_benchmark.yml [hunk 3]
```diff
@@ -107,6 +107,12 @@ jobs:
         run: |
           make BENCH_OP_FLAVOR=${{ matrix.op_flavor }} bench_${{ matrix.command }}
 
+      # Run these benchmarks only once
+      - name: Run OPRF benchmarks
+        if: matrix.command == 'integer' && matrix.op_flavor == 'default'
+        run: |
+          make bench_integer_oprf
```
Comment (IceTDrinker) [inline .github/workflows/integer_full_benchmark.yml:114]: could not put the oprf in the same benchmarks as the integers ?

sorry just realized now

Comment (soonum) [reply .github/workflows/integer_full_benchmark.yml:114]: You mean like the cast benchmarks in shortint ?

Comment (IceTDrinker) [reply .github/workflows/integer_full_benchmark.yml:114]: what do you mean ?

Comment (soonum) [reply .github/workflows/integer_full_benchmark.yml:114]: What do YOU mean then :smirk: ?
When you say `put the oprf in the same benchmark as the integers` . At the workflow level ? Makefile level ? Rust benchmark file level ?

Comment (IceTDrinker) [reply .github/workflows/integer_full_benchmark.yml:114]: rust benchmark, sorry wasn't clear

File: tfhe/benches/integer/oprf.rs [hunk 4]
```diff
@@ -1,33 +1,75 @@
...
-    });
+        Some((param, num_block, bit_size))
+    }
 }
 
-criterion_group!(oprf2, oprf);
+pub fn unsigned_oprf(c: &mut Criterion) {
+    let bench_name = "integer::unsigned_oprf";
+
+    let mut bench_group = c.benchmark_group(bench_name);
```
Comment (IceTDrinker) [inline tfhe/benches/integer/oprf.rs:44]: any reason to keep it in a separate module ?

Comment (soonum) [reply tfhe/benches/integer/oprf.rs:44]: The file `integer/bench.rs` is already quite big so I wanted to keep OPRF benchmarks away. 
This is the only reason.

Comment (IceTDrinker) [reply tfhe/benches/integer/oprf.rs:44]: Does this work ? I see the utilities mod done a bit differently ?

Comment (IceTDrinker) [reply tfhe/benches/integer/oprf.rs:44]: I don’t know if the params/number of bits correspond to what we do in the other file ?

**Pull request merged**